### PR TITLE
Add acceptance test cleanup

### PR DESCRIPTION
**Pull request by curtbushko**
### Changes proposed in this PR ###  
- We often have acceptance tests that fail because resources are left behind. At this time I do not trust that helm delete is deleting everything it created.
- This is a first pass attempt to nuke all the helm things for a release.

### How I've tested this PR ###

- Ran through acceptance tests (all green)

### How I expect reviewers to test this PR ###

👀 

### Checklist ###
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by wilkermichael** [comment]
Just a heads up that we're entering a code-freeze today at some point (for the backports) so maybe hold off on merging until you check with @t-eckert

File: acceptance/framework/consul/helm_cluster.go
Comment (wilkermichael) [inline acceptance/framework/consul/helm_cluster.go:227]: suggestion: Just a thought but might be worth having a function or something that we call that just generates `"release="+h.releaseName` since it's used everywhere. Less chance of accidentally mispelling/creating wrong string in the future.

File: acceptance/framework/consul/helm_cluster.go
Comment (wilkermichael) [inline acceptance/framework/consul/helm_cluster.go:351]: are there any steps we can take if these resources still exist?

File: acceptance/framework/consul/helm_cluster.go
Comment (wilkermichael) [inline acceptance/framework/consul/helm_cluster.go:355]: Good idea to verify resource deletion... this will hopefully help us diagnose which resource is causing things not to be deleted properly in the future.

**Review by wilkermichael — APPROVED** [review]
Awesome job!

I am hopeful that this will help with some of those "resources still exist" errors 🤞

**Comment by curtbushko** [comment]
I will wait until after code freeze to merge this just so that we do not cause any problems.

**Comment by t-eckert** [comment]
> I will wait until after code freeze to merge this just so that we do not cause any problems.

Ack.

**Pull request merged**
